File: Clase__74__14-10__Fragments/Fragments/app/src/main/java/com/example/ejerciciofragmentado/fragmentos/ListFragment.java
package com.example.ejerciciofragmentado.fragmentos;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.ejerciciofragmentado.MainActivity;
import com.example.ejerciciofragmentado.R;
import com.example.ejerciciofragmentado.adapter.ProductoAdapter;
import com.example.ejerciciofragmentado.databinding.FragmentAddBinding;
import com.example.ejerciciofragmentado.databinding.FragmentListBinding;
import com.example.ejerciciofragmentado.model.Producto;

import java.util.ArrayList;


public class ListFragment extends Fragment {
    FragmentListBinding binding;
    private RecyclerView recyclerView;



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentListBinding.inflate(inflater, container, false);
        View view = binding.getRoot();

        ArrayList<Producto> listadoProductos = (ArrayList<Producto>)
                getArguments().getSerializable("listaProducto");

        recyclerView = binding.recycler;

        ProductoAdapter adapter = new ProductoAdapter(listadoProductos);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setAdapter(adapter);





        return view;


    }
}

File: Clase__86__02-11__MVP/ContactosApp/app/src/main/java/com/example/simplecontactos/fragments/ui/ListFragment.kt
package com.example.simplecontactos.fragments.ui

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.activityViewModels
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.simplecontactos.adapter.ContactoAdapter
import com.example.simplecontactos.databinding.FragmentListBinding
import com.example.simplecontactos.presenter.home.HomePresenter
import com.example.simplecontactos.presenter.home.HomePresenterImpl
import com.example.simplecontactos.viewmodel.ContactoViewModel

class ListFragment : Fragment() {
   private lateinit var binding: FragmentListBinding

   private lateinit var presenter: HomePresenter

   val viewModel : ContactoViewModel by activityViewModels()



    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
       binding = FragmentListBinding.inflate(layoutInflater, container, false)

        //val presenter = HomePresenterImpl()

        val adapter = ContactoAdapter(requireActivity())

        with(binding){
            recycler.layoutManager = LinearLayoutManager(context)
            recycler.adapter = adapter

        }







//        viewModel.listaContacto.observe(viewLifecycleOwner, {
//            adapter.setContacto(it)
//        })




        return binding.root

    }


}